### PR TITLE
Create an example of ReadRows()

### PR DESCRIPTION
**Pull request by coryan**
Refactored some common code in the upload_taq* examples and create a new example showing how to use the ReadRows() API.

### CODE REVIEW

File: bigtable/api/read_rows.cc [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+  bigtable::ReadRowsResponse response;
+  while (stream->Read(&response)) {
+    for (auto& cell_chunk : *response.mutable_chunks()) {
+      if (not cell_chunk.row_key().empty()) {
+	current_row_key = cell_chunk.row_key();
+      }
+      if (cell_chunk.has_family_name()) {
+	current_column_family = cell_chunk.family_name().value();
+	if (current_column_family != "taq") {
+	  throw std::runtime_error("strange, only 'taq' family name expected in the query");
```
Comment (garye) [inline bigtable/api/read_rows.cc]: maybe add a `family_name_regex_filter`? Might still have the assertions but would be good for an example.

File: bigtable/api/read_rows.cc [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+	}
+      }
+      if (cell_chunk.timestamp_micros() != 0) {
+	throw std::runtime_error("strange, only the 0 timestamp expected in the query");
+      }
+      current_timestamp = cell_chunk.timestamp_micros();
+      chunks.emplace_back(std::move(*cell_chunk.mutable_value()));
+      if (cell_chunk.commit_row()) {
+	// ... process this cell, we want to convert the sequence of
+	// chunks into a Quotes proto, to do so we need to create the
```
Comment (garye) [inline bigtable/api/read_rows.cc]: Does it make things easier if you use `value_size`, if present, to pre-allocate a buffer that will fit the cell's entire value? If not present you can assume it's not chunked.

File: bigtable/api/read_rows.cc [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+
+#include <algorithm>
+#include <ciso646>
+#include <deque>
+#include <fstream>
+#include <sstream>
+#include <thread>
+
+// We want to show the correct way to receive a large Bigtable cell
+// value in ReadRows().  Bigtable can break down such cell values
```
Comment (coryan) [inline bigtable/api/read_rows.cc]: Done.

File: bigtable/api/read_rows.cc (hunk 1 above)
Comment (coryan) [inline bigtable/api/read_rows.cc]: Done.

**Comment by coryan** [comment]
PTAL.

File: bigtable/api/read_rows.cc [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+	namespace io = google::protobuf::io;
+	// ... first create a vector with a zero copy stream for each
+	// chunk we received, user a smart pointer to automatically
+	// delete them ...
+	std::vector<std::shared_ptr<io::ZeroCopyInputStream>> streams_holder;
+	streams_holder.reserve(chunks.size());
+	for (auto const& c : chunks) {
+	  streams_holder.push_back(std::make_shared<io::ArrayInputStream>(c.data(), c.size()));
+	}
+	// ... then put the raw pointers into a contiguous buffer,
```
Comment (garye) [inline bigtable/api/read_rows.cc]: Does it make things easier if you use `value_size`, if present, to pre-allocate a buffer that will fit the cell's entire value? If not present you can assume it's not chunked.

File: bigtable/api/read_rows.cc [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+
+  std::unique_ptr<bigtable::Bigtable::Stub> bt_stub(
+      bigtable::Bigtable::NewStub(channel));
+
+  // ... once the table is created and the data uploaded we can make
+  // the query ...
+  bigtable::ReadRowsRequest request;
+  request.set_table_name(table_name);
+
+  // ... show how to use filters, there are probably several ways to do this ...
```
Comment (coryan) [inline bigtable/api/read_rows.cc]: Done.

File: bigtable/api/read_rows.cc (hunk 4 above)
Comment (coryan) [inline bigtable/api/read_rows.cc]: Done.  I think we will also need an example with the ZeroCopy objects, but you are right, the first example should be simpler.

**Pull request merged**